tests/integration/allocator_accessibility.cpp

```diff
@@ -0,0 +1,174 @@
+//
+// SPDX-License-Identifier: (MIT)
+//////////////////////////////////////////////////////////////////////////////
+#include "gtest/gtest.h"
+#include "umpire/Allocator.hpp"
+#include "umpire/ResourceManager.hpp"
+#include "umpire/Umpire.hpp"
+#include "umpire/config.hpp"
+#include "umpire/util/MemoryResourceTraits.hpp"
+
+using cPlatform = camp::resources::Platform;
+using myResource = umpire::MemoryResourceTraits::resource_type;
+
+struct host_platform {};
+
+template <typename Platform>
+struct allocate_and_use{};
+
+template<>
+struct allocate_and_use<host_platform>
+{
+  void test(umpire::Allocator* alloc, size_t size)
+  {
+    size_t* data = static_cast<size_t*>(alloc->allocate(size * sizeof(size_t)));
+    data[0] = size * size;
+    alloc->deallocate(data);
+  }
+};
+
+#if defined(UMPIRE_ENABLE_CUDA) || defined(UMPIRE_ENABLE_HIP)
+__global__ void tester(size_t* d_data, size_t size)
+{
+  int idx = blockDim.x * blockIdx.x + threadIdx.x;
+   
+  if (idx == 0) {
+    d_data[0] = size * size;
+  }
+}
+#endif
+
+#if defined(UMPIRE_ENABLE_CUDA)
+struct cuda_platform {};
+
+template<>
+struct allocate_and_use<cuda_platform>
+{
+  void test(umpire::Allocator* alloc, size_t size)
+  {
+    size_t* data = static_cast<size_t*>(alloc->allocate(size * sizeof(size_t)));
+    tester<<<1, 16>>>(data, size);
+    cudaDeviceSynchronize();
+    alloc->deallocate(data);
+  }
+};
+#endif
+
+#if defined(UMPIRE_ENABLE_HIP)
+struct hip_platform{};
+
+template<>
+struct allocate_and_use<hip_platform>
+{
+  void test(umpire::Allocator* alloc, size_t size)
+  {
+    size_t* data = static_cast<size_t*>(alloc->allocate(size * sizeof(size_t)));
+    hipLaunchKernelGGL(tester, dim3(1), dim3(16), 0,0, data, size);
+    hipDeviceSynchronize();
+    alloc->deallocate(data);
+  }
+};
+#endif
+
+#if defined(UMPIRE_ENABLE_OPENMP_TARGET)
+struct omp_target_platform{};
+
+template<>
+struct allocate_and_use<omp_target_platform>
+{
+  void test(umpire::Allocator* alloc, size_t size)
+  {
+    int dev = alloc->getAllocationStrategy()->getTraits().id;
+    size_t* data = static_cast<size_t*>(alloc->allocate(size * sizeof(size_t)));
+    size_t* d_data{static_cast<size_t*>(data)};
+
+#pragma omp target is_device_ptr(d_data) device(dev)
+#pragma omp teams distribute parallel for schedule(static, 1)
+    for (auto i = 0; i < size; ++i) {
+      d_data[i] = static_cast<size_t>(i);
+    }
+
+    alloc->deallocate(data);
+  }
+};
+#endif
+
+class AllocatorAccessibilityTest : public ::testing::TestWithParam<std::string> {
+ public:
+  virtual void SetUp()
+  {
+    auto& rm = umpire::ResourceManager::getInstance();
+    m_allocator = new umpire::Allocator(rm.getAllocator(GetParam()));
+  }
+
+  virtual void TearDown()
+  {
+    m_allocator->release();
+    delete m_allocator;
+  }
+  
+  umpire::Allocator* m_allocator;
+  size_t m_size = 42;
+};
+
+TEST_P(AllocatorAccessibilityTest, AccessibilityFromPlatform)
+{
+  if(is_accessible(cPlatform::host, *m_allocator)) {
```

Comment:
Similar question to another comment made in a different file.  Shouldn't this have a namespace qualified?  I'm confused why it compiles.
```suggestion
  if(umpire::is_accessible(cPlatform::host, *m_allocator)) {
```

Reply:
The umpire.hpp file is included, I assume that's why this works. The is_accessible function is not a member of the umpire namespace, so if I did something like that, it'd have to make it something like in line 14 or 15. Am I explaining this right? I feel like I'm making this confusing, but do you see what I'm talking about?

Reply:
Oh, yes.  I get it.  I think that these functions need to be part of the umpire namespace.

Reply:
The function is in the `umpire` namespace. I think there might a `using namespace umpire` line somewhere that makes this work.

Reply:
Adding a `using` declaration like on line 14 or 15 is not necessary if you explicitly write out the fully qualified name.

Reply:
If there is a `using umpire` statement somewhere, I would like to get it removed (especially if it is in a header).  using namespaces should be discouraged in libraries.

Reply:
Oh, well Umpire.hpp has the umpire namespace, maybe when I include that file, it kinda gets added in I guess

Reply:
uh oh.  IMHO, that is a problem.